connection.go

```diff
@@ -100,7 +100,7 @@ calls to notify it of various events.
 */
 type Connection struct {
 	handler            ConnectionHandler
-	role               uint8
+	Role               Role
```

Comment:
Please leave this private, like everything else here. It's part of the public API and so we don't want people frobbing it. If you need it, provide an accessor.